src/core/execute.c

```diff
@@ -3189,6 +3189,24 @@ static int apply_mount_namespace(
                 }
         }
 
+        if (MANAGER_IS_SYSTEM(u->manager)) {
+                propagate_dir = path_join("/run/systemd/propagate/", u->id);
+                if (!propagate_dir)
+                        return -ENOMEM;
+                incoming_dir = strdup("/run/host/incoming");
+                if (!incoming_dir)
+                        return -ENOMEM;
+        } else if (context->protect_home == PROTECT_HOME_NO || context->protect_home == PROTECT_HOME_TMPFS) {
```

Comment:
hmm, have you tested this? do we actually support ProtectHome= in the --user instance?  I am surprised this is supposed to work.

IN_SET() is your friend.

Reply:
Fixed IN_SET - it was a while ago, but IIRC a test was failing if I tried to set it up unconditionally. Even if not supported right now, probably safer to have it in case it ProtectHome=read-only becomes supported in the future? Or would you rather have me remove it entirely?

Reply:
reason being, if we can't create the XDG stuff it cannot possibly work

Reply:
remove

Reply:
ok, removed the check for protect_home* - PTAL